test/solver/gmres_kernels.cpp

```diff
@@ -272,6 +273,50 @@ TEST_F(Gmres, GmresKernelMultiAxpyIsEquivalentToRef)
     GKO_ASSERT_ARRAY_EQ(stop_status, d_stop_status);
 }
 
+TEST_F(Gmres, GmresKernelMultiDotIsEquivalentToRef)
+{
+    initialize_data();
+
+    auto krylov_basis = krylov_bases->create_submatrix(
+        gko::span{
+            0, x->get_size()[0] * (gko::solver::gmres_default_krylov_dim - 1)},
+        gko::span{0, x->get_size()[1]});
+    auto d_krylov_basis = d_krylov_bases->create_submatrix(
+        gko::span{0, d_x->get_size()[0] *
+                         (gko::solver::gmres_default_krylov_dim - 1)},
+        gko::span{0, d_x->get_size()[1]});
+    auto next_krylov = krylov_bases->create_submatrix(
+        gko::span{
+            x->get_size()[0] * (gko::solver::gmres_default_krylov_dim - 1),
+            x->get_size()[0] * gko::solver::gmres_default_krylov_dim},
+        gko::span{0, x->get_size()[1]});
+    auto d_next_krylov = d_krylov_bases->create_submatrix(
+        gko::span{
+            d_x->get_size()[0] * (gko::solver::gmres_default_krylov_dim - 1),
+            d_x->get_size()[0] * gko::solver::gmres_default_krylov_dim},
+        gko::span{0, d_x->get_size()[1]});
+    gko::kernels::reference::gmres::multi_dot(
+        ref, krylov_basis.get(), next_krylov.get(), hessenberg_iter.get());
+    gko::kernels::GKO_DEVICE_NAMESPACE::gmres::multi_dot(
+        exec, d_krylov_basis.get(), d_next_krylov.get(),
+        d_hessenberg_iter.get());
+
+    // The multidot computation does not set the value below the diagonal
+    // in the Hessenberg matrix column(s), as that is done after the
+    // orthogonalization of the next basis vector. In this test, we
+    // are checking the column(s) created on the last iteration before the
+    // solver's restart would be triggered, so it is only the final row of
+    // the Hessenberg column(s) that we ignore.
```

Comment:
Yeah, good point. Originally, I thought it would be better to have this specifically marked/explained for future reference, but you're right, it's just unnecessarily complicated. I'll take it out.